### PR TITLE
Comparing collection using equals method for assertEquals

### LINKED ISSUE #2296: assertEquals doesn't work for sets
### TestNG Version
7.1.0

### Expected behavior
`assertEquals` yields true if called with two sets that contain the same elements, even if they are defined as `Collection`.

### Actual behavior
Depending on the type of set and the type of objects they contain `assertEquals` might work well or fail because it relies on iterators to compare the two sets and in the case of sets the iteration order is not guaranteed. This behavior is is most evident on `LinkedHashSet`s but also appears on `HashSet`s depending on the type of objects they contain. I suspect it works well with `TreeSet`s, which have a predictable iteration order.

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample
Test case excerpt (full test in attachment):

```
    @Test
    public void assertEqualsWorksOnLinkedHashSets() {
        final Collection<Integer> oneSet = new LinkedHashSet<>();
        oneSet.add(1);
        oneSet.add(2);
        final Collection<Integer> anotherSet = new LinkedHashSet<>();
        anotherSet.add(2);
        anotherSet.add(1);
        assertTrue(oneSet.equals(anotherSet)); //passes
        assertEquals(oneSet, anotherSet); //fails
    }
```

[AssertEqualsOnSetsTest.zip](https://github.com/cbeust/testng/files/4485658/AssertEqualsOnSetsTest.zip)

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2296 
To check whether the two objects are equal, have replaced all the comparison operator (=) with equals method for non primitive data types. This will add capability to compare the sets as they do not always guarantee the insertion order.

### CODE REVIEW

**Review by juherr — CHANGES_REQUESTED** [review]
Please check everywhere `actual == expected` is used.
Especially https://github.com/cbeust/testng/blame/master/src/main/java/org/testng/Assert.java#L1352

**Comment by prashant-maroti** [comment]
@juherr @krmahadevan Have replaced all the occurrences of comparison operator (=) with 'equals' method for non primitive data types.

File: src/main/java/org/testng/Assert.java [hunk 1]
```diff
@@ -1210,7 +1210,7 @@ public static void assertEquals(Iterable<?> actual, Iterable<?> expected, String
    * @param message  the assertion error message
    */
   public static void assertEquals(Object[] actual, Object[] expected, String message) {
-    if (actual == expected) {
+    if (Objects.equals(actual, expected)) {
```
Comment (juherr) [inline src/main/java/org/testng/Assert.java:1213]: `Arrays.equals` looks better here.

File: src/test/java/org/testng/AssertTest.java [hunk 2]
```diff
@@ -41,6 +41,13 @@ public void nullSetAssertEquals() {
     Assert.assertEquals(actual, expected);
   }
 
+  @Test
+  public void testSetAssertEquals() {
```
Comment (juherr) [inline src/test/java/org/testng/AssertTest.java:45]: I think you are using `assertEquals(Collection)` in the test.
But as `assertEquals(Set)` and `assertEquals(Collection)` are different, could you add a test for both methods?

**Review by juherr — CHANGES_REQUESTED** [review]

File: src/main/java/org/testng/Assert.java (hunk 1 above)
Comment (prashant-maroti) [reply src/main/java/org/testng/Assert.java]: Have replaced this with Arrays.equals.

File: src/test/java/org/testng/AssertTest.java (hunk 2 above)
Comment (prashant-maroti) [reply src/test/java/org/testng/AssertTest.java:60]: I have added test cases for list, set and collection having comparison with there equals implementation.

**Review by juherr — APPROVED** [review]

**Comment by prashant-maroti** [comment]
@krmahadevan Kindly review. Have resolved the conflicts with master.

**Review by krmahadevan — APPROVED** [review]

**Pull request merged**